libflux/src/semantic/nodes.rs

```diff
@@ -149,12 +217,35 @@ pub struct ReturnStmt {
 
 #[derive(Debug, PartialEq, Clone)]
 pub struct VariableAssgn {
+    free: Tvars,
+    cons: Option<TvarKinds>,
+
     pub loc: ast::SourceLocation,
 
     pub id: Identifier,
     pub init: Expression,
 }
 
+impl VariableAssgn {
+    pub fn new(id: Identifier, init: Expression, loc: ast::SourceLocation) -> VariableAssgn {
+        VariableAssgn {
+            free: Tvars::new(),
+            cons: None,
+            loc,
+            id,
+            init,
+        }
+    }
+}
+
+pub fn poly_type_of(v: &VariableAssgn) -> PolyType {
```

Comment:
Should this be a method on the VariableAssgn type? That way it is easier to discover its existence?

Reply:
I was initially thinking it read a little better as `semantic::poly_type_of(var)` vs `var.poly_type_of()` but I don't have a strong opinion. I can change it to make it easier to discover.

Reply:
I am not sure either. I guess lets leave it for now and we can reconsider later.